core/src/main/java/org/modelmapper/spi/PropertyReferenceMapping.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modelmapper.spi;
+
+/**
+ * Represents mapping actions, to set a destination property from a source property,
+ * or to skip a destination property.
+ *
+ * @param <S> source type
+ * @param <D> destination type
+ *
+ * @author Chun Han Hsiao
+ */
+public interface PropertyReferenceMapping<S, D> {
```

Comment:
Similar to `TypeMapMapping` being renamed `PropertyMapper`, I would name this something with `Mapper` in it like maybe `PropertyReferenceMapper` since it's used to create mappings.